File: app/schemas/planning.py
from __future__ import annotations

from datetime import datetime
from typing import Optional

from pydantic import BaseModel, ConfigDict, Field

from app.schemas.brand import BrandProfileResponse


class PlanningCompanyMatchResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    review_status: str
    confidence: float
    matched_alias: str
    matched_field: str
    excerpt: str
    detector_version: str
    first_seen_at: datetime
    last_seen_at: datetime
    brand: BrandProfileResponse


class PlanningRecordResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    source_id: str
    external_record_id: str
    event_type: str
    stage: Optional[str]
    title: str
    summary: Optional[str]
    evidence_excerpt: Optional[str]
    agenda_item_number: Optional[str]
    meeting_name: Optional[str]
    governing_body: Optional[str]
    project_name: Optional[str]
    address: Optional[str]
    city: Optional[str]
    state: Optional[str]
    postal_code: Optional[str]
    parcel_id: Optional[str]
    jurisdiction: Optional[str]
    applicant_name: Optional[str]
    owner_name: Optional[str]
    developer_name: Optional[str]
    latitude: Optional[float]
    longitude: Optional[float]
    meeting_at: Optional[datetime]
    published_at: Optional[datetime]
    decision_at: Optional[datetime]
    source_url: Optional[str]
    signal_categories: list[str] = Field(default_factory=list)
    priority_reasons: list[str] = Field(default_factory=list)
    priority_score: float
    confidence: float
    first_seen_at: datetime
    last_seen_at: datetime
    company_matches: list[PlanningCompanyMatchResponse] = Field(default_factory=list)


class PlanningCompanyMatchReview(BaseModel):
    review_status: str = Field(pattern=r"^(candidate|confirmed|dismissed)$")

File: src/types/planning.ts
import type { BrandProfile } from './brand';

export interface PlanningCompanyMatch {
  id: string;
  raw_record_id: string;
  review_status: string;
  confidence: number;
  matched_alias: string;
  matched_field: string;
  excerpt: string;
  detector_version: string;
  first_seen_at: string;
  last_seen_at: string;
  brand: BrandProfile;
}

export interface PlanningRawEvidence {
  id: string;
  external_record_id: string;
  content_hash: string;
  source_updated_at?: string | null;
  received_at: string;
}

export interface PlanningRecord {
  id: string;
  source_id: string;
  external_record_id: string;
  reference_number?: string | null;
  event_type: string;
  stage?: string | null;
  title: string;
  summary?: string | null;
  evidence_excerpt?: string | null;
  agenda_item_number?: string | null;
  meeting_name?: string | null;
  governing_body?: string | null;
  project_name?: string | null;
  address?: string | null;
  city?: string | null;
  state?: string | null;
  postal_code?: string | null;
  parcel_id?: string | null;
  jurisdiction?: string | null;
  applicant_name?: string | null;
  owner_name?: string | null;
  developer_name?: string | null;
  latitude?: number | null;
  longitude?: number | null;
  meeting_at?: string | null;
  published_at?: string | null;
  decision_at?: string | null;
  source_url?: string | null;
  signal_categories: string[];
  priority_reasons: string[];
  priority_score: number;
  confidence: number;
  first_seen_at: string;
  last_seen_at: string;
  latest_raw_record: PlanningRawEvidence;
  company_matches: PlanningCompanyMatch[];
}

export interface PlanningSignalParams {
  brand_id?: string;
  state?: string;
  city?: string;
  category?: string;
  minimum_priority?: number;
  limit?: number;
}
